### PR TITLE
drivers: sensor: amg88xx: Update driver to use new driver mechanisms

### PR DESCRIPTION
**Pull request by galak**
Move driver to use gpio_dt_spec and DEVICE_DT_GET for handling access
to bus parent device struct.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
The I2C and GPIO changes should be in separate commits.

Also, why did you rename i2c to i2c_bus_dev? The change adds some unnecessary noise to the diff

File: drivers/sensor/amg88xx/amg88xx_trigger.c [hunk 1]
```diff
@@ -165,24 +163,26 @@ int amg88xx_init_interrupt(const struct device *dev)
...
+		if (config->int_gpio.port) {
+			LOG_ERR("%s: device %s is not ready", dev->name,
+						config->int_gpio.port->name);
+			return -ENODEV;
+		}
 
-	drv_data->gpio_pin = config->gpio_pin;
+		LOG_DBG("%s: int_gpio not defined in DT", dev->name);
+		return 0;
+	}
```
Comment (gmarull) [inline drivers/sensor/amg88xx/amg88xx_trigger.c:177]: Driver uses `GPIO_DT_SPEC_INST_GET`, not `GPIO_DT_SPEC_INST_GET_OR`, meaning `int_gpio.port` can't be NULL. We should also return `-ENODEV`.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/amg88xx/amg88xx.h [hunk 2]
```diff
@@ -80,10 +78,10 @@ struct amg88xx_data {
 
 #ifdef CONFIG_AMG88XX_TRIGGER
 	const struct device *dev;
-	const struct device *gpio;
-	uint8_t gpio_pin;
+	const struct amg88xx_config *cfg;
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/sensor/amg88xx/amg88xx.h]: removed.  getting `config` from `dev` pointer now.

File: samples/sensor/amg88xx/src/main.c [hunk 3]
```diff
@@ -51,11 +51,10 @@ void print_buffer(void *ptr, size_t l)
 void main(void)
 {
 	int ret;
-	const struct device *dev = device_get_binding(
-				DT_LABEL(DT_INST(0, panasonic_amg88xx)));
+	const struct device *dev = DEVICE_DT_GET(DT_INST(0, panasonic_amg88xx));
```
Comment (gmarull) [inline samples/sensor/amg88xx/src/main.c:54]:
```suggestion
	const struct device *dev = DEVICE_DT_GET_ONE(panasonic_amg88xx);
```

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**